adapters/arangodb/src/adapter.ts

```diff
@@ -198,7 +198,7 @@ export class ArangoAdapter extends AdapterBase {
         if (
             this.collections.find(
                 c =>
-                    c.database === modelInfo.database &&
+                    (c.database != null && c.database === modelInfo.database) &&
```

Comment:
I think we'd want to check with the default database name. Otherwise you could have two models named the same one in the default database, and one in one you specified and then only one would be created

something like:
```ts
const dbName = modelInfo.database || this.defaultDatabase;
if (
  this.collections.find(c =>
    c.database == dbName &&
    c.name == modelInfo.table,
  ) == null
) {
  // ...
}
```

Reply:
Likewise, I guess this should use the dbName var also

```typescript
        const collection = this.db(modelInfo.database).collection(
            modelInfo.table,
        );
```

But the TS check would assume a fail as `defaultDatabase` can be undefined according to types. Which fails on the method `db(model: String)` as it's expecting a string _only_.

Reply:
The `db` method handles the use of the default there

Reply:
Okay updated the PR with some changes.